src/lib/libast/misc/procopen.c

```diff
@@ -485,12 +485,33 @@ Proc_t *procopen(const char *cmd, char **argv, char **envv, long *modv, int flag
         }
         if (flags & (PROC_DAEMON | PROC_SESSION)) modify(proc, forked, PROC_sys_pgrp, -1, 0);
         if (forked || (flags & PROC_OVERLAY)) {
+            int status = 0;
             if ((flags & PROC_PRIVELEGED) && !geteuid()) {
-                setuid(geteuid());
-                setgid(getegid());
+                while (((status = setuid(geteuid())) < 0) && (errno == EAGAIN));
+                if (status < 0) {
+                    error(ERROR_system(0), "setuid(%d) failed", geteuid());
+                    return NULL;
```

Comment:
I can see reporting a non-fatal error but is it okay to return early? You're changing the behavior with the insertion of these `return` statements and it isn't obvious that is safe to do.

Reply:
According to `src/lib/libast/include/proc.h`:
```
#define PROC_PRIVELEGED (1 << 9)  /* setuid(0), setgid(getegid())       */
```
This flag is set to execute a new process under a specific user id and group id. If we fail to honour it, I presume we should return back.

Reply:
I'm a little uncomfortable because while this change seems correct it is changing the logic.